lms/djangoapps/teams/views.py

```diff
@@ -1143,6 +1149,39 @@ def get(self, request):
                 return Response(status=status.HTTP_404_NOT_FOUND)
             if not has_specific_team_access(request.user, team):
                 return Response(status=status.HTTP_403_FORBIDDEN)
+        elif 'teamset_id' in request.query_params:
+            if 'course_id' not in request.query_params:
+                return Response(
+                    build_api_error(ugettext_noop("teamset_id requires course_id to also be provided.")),
+                    status=status.HTTP_400_BAD_REQUEST
+                )
+
+            if not has_team_api_access(request.user, requested_course_key):
+                return Response(status=status.HTTP_404_NOT_FOUND)
+
+            course_module = modulestore().get_course(requested_course_key)
+            if not course_module:
+                return Response(status=status.HTTP_404_NOT_FOUND)
+            specified_username_or_team = True
+            teamsets = course_module.teams_configuration.teamsets_by_id
+            teamset_id = request.query_params['teamset_id']
+            teamset = teamsets.get(teamset_id, None)
+            if not teamset:
+                return Response(
+                    build_api_error(ugettext_noop("No teamset found in given course with given id")),
+                    status=status.HTTP_404_NOT_FOUND
+                )
+            teamset_teams = CourseTeam.objects.filter(course_id=requested_course_key, topic_id=teamset_id)
+            teams_with_access = list(filter(
+                lambda team: has_specific_team_access(request.user, team),
+                teamset_teams
+            ))
+            if not teams_with_access:
+                return Response(
+                    build_api_error(ugettext_noop("No teamset found in given course with given id")),
```

Comment:
Copypasta. No 'teams' instead of 'teamset' ?

Reply:
nope, this was on purpose. if you put in a teamset that exists but you shouldn't be able to know about, we want to act like it doesn't exist